src/Compiler/CMakeLists.txt

```diff
@@ -66,10 +66,49 @@ add_onnx_mlir_library(CompilerPasses
   )
 
 if (ONNX_MLIR_VENDOR)
-  set_source_files_properties(CompilerUtils.cpp
-    PROPERTIES COMPILE_DEFINITIONS "ONNX_MLIR_VENDOR=\"${ONNX_MLIR_VENDOR} \"")
+  list(APPEND DEFINITIONS "ONNX_MLIR_VENDOR=\"${ONNX_MLIR_VENDOR}\"")
 endif()
 
+# PRODUCT_VERSION is specified/cached.
+if (PRODUCT_VERSION)
+  string(REGEX MATCH "^([0-9]+)(\\.([0-9]+))?(\\.([0-9]+))?(\\-(.+))$"
+	  VERSION_MATCH ${PRODUCT_VERSION})
+
+  # For a version string 1.2.3-foo,
+  # CMAKE_MATCH_1 = 1
+  # CMAKE_MATCH_3 = 2
+  # CMAKE_MATCH_5 = 3
+  # CMAKE_MATCH_7 = foo
+  set(PRODUCT_VERSION_MAJOR ${CMAKE_MATCH_1})
+  set(PRODUCT_VERSION_MINOR ${CMAKE_MATCH_3})
+  set(PRODUCT_VERSION_PATCH ${CMAKE_MATCH_5})
+  set(PRODUCT_ID ${CMAKE_MATCH_7})
+
+  message(DEBUG "PRODUCT_VERSION_MAJOR = ${PRODUCT_VERSION_MAJOR}")
+  message(DEBUG "PRODUCT_VERSION_MINOR = ${PRODUCT_VERSION_MINOR}")
+  message(DEBUG "PRODUCT_VERSION_PATCH = ${PRODUCT_VERSION_PATCH}")
+  message(DEBUG "PRODUCT_ID = ${PRODUCT_ID}")
+
+  # Major and product ID are mandatory, minor and patch are optional.
+  # Major must be greater than zero.
+  # When not specified, minor and patch are empty.
+  if (("${PRODUCT_VERSION_MAJOR}" STREQUAL "")
+	  OR ("${PRODUCT_ID}" STREQUAL "")
+	  OR (PRODUCT_VERSION_MAJOR LESS_EQUAL 0))
```

Comment:
There is no need to check for PRODUCT_VERSION_MAJOR < 0. The regex match does not allow that.

Reply:
Changed `PRODUCT_VERSION_MAJOR LESS_EQUAL 0` to `PRODUCT_VERSION_MAJOR EQUAL 0`.

Reply:
0 is a valid major I'd think?

Reply:
I am following the rules in https://github.com/onnx/onnx-mlir/pull/1409#discussion_r867151399:
> if PRODUCT_ID is set at least `PRODUCT_VERSION_MAJOR` must be set to a positive integer value different that 0

@etiotto is 0 a valid major?

Reply:
It should be, lots of products ship as 0.0.x, or 0.x.y, etc.

Reply:
Yes 0 is a valid major IMO

Reply:
Changed to allow 0.